src/py/flwr/common/serde.py

```diff
@@ -29,16 +29,148 @@
 
 from . import typing
 
-# pylint: disable=missing-function-docstring
+#  === ServerMessage message ===
+
+
+def server_message_to_proto(server_message: typing.ServerMessage) -> ServerMessage:
+    """Serialize `ServerMessage` to ProtoBuf."""
+    if server_message.get_properties_ins is not None:
+        return ServerMessage(
+            get_properties_ins=get_properties_ins_to_proto(
+                server_message.get_properties_ins,
+            )
+        )
+    if server_message.get_parameters_ins is not None:
+        return ServerMessage(
+            get_parameters_ins=get_parameters_ins_to_proto(
+                server_message.get_parameters_ins,
+            )
+        )
+    if server_message.fit_ins is not None:
+        return ServerMessage(
+            fit_ins=fit_ins_to_proto(
+                server_message.fit_ins,
+            )
+        )
+    if server_message.evaluate_ins is not None:
+        return ServerMessage(
+            evaluate_ins=evaluate_ins_to_proto(
+                server_message.evaluate_ins,
+            )
+        )
+    raise Exception("No instruction set in ServerMessage, cannot serialize to ProtoBuf")
```

Comment:
I'm not sure what you mean. The patterns are slightly different because we have `OneOf` in the ProtoBuf message and a set of optional fields in the Python class, so there's a semantic difference between the two representations. In the ProtoBuf `OneOf` case we don't consider the "no message" case at all.

Reply:
@danieljanes you are right. My bad. I kinda missed that this converts to proto, and the `dataclass` can hold all messages.